docs/vector-library.md

```diff
@@ -0,0 +1,129 @@
+
+## Design
+
+The default metatable for vectors should now be the `vector` library. While buffers and coroutines do not have this, vectors likely should for ergonomics.
+
+### Library functions & constants
+
+This RFC proposes the following basic functions & constants as a starting point, but as with all builtin libraries, future RFCs can propose additional functions.
+
+---
+
+`vector(x: number, y: number, z: number)`
+
+Creates a vector with 3 components: x, y, z. If the `LUA_VECTOR_SIZE` configuration option is set to 4, a fourth argument `w: number?` will be introduced.
+
+Due to the common usage of vectors, vector creation should be ergonomic. Therefore, it is probably worth breaking the `create()` naming standard.
+
+`vector.magnitude(vec: vector): number`
+
+Calculates the magnitude of a given vector.
+
+`vector.normalized(vec: vector): vector`
+
+Returns the normalized version (aka unit vector) of a given vector.
+
+`vector.cross(vec1: vector, vec2: vector): vector`
+
+Returns the cross product of two vectors. If 4-wide vectors are enabled, this function will ignore the fourth component, and return the 3-dimensional cross product.
+
+`vector.dot(vec1: vector, vec2: vector): number`
+
+Returns the dot product of two vectors.
+
+`vector.floor(vec: vector): vector`
+
+Equivalent of `math.floor` for vectors.
+
+`vector.ceil(vec: vector): vector`
+
+Equivalent of `math.ceil` for vectors.
+
+`vector.angle(vec1: vector, vec2: vector, axis: vector?): number`
+
+Returns the angle between two vectors in radians. The axis, if specified, is used to determine the sign of the angle.
+
+`vector.abs(vec: vector): vector`
+
+Applies math.abs component-wise for each vector. Equivalent of: `vector(math.abs(vec.x), ect)`.
+
+`vector.sign(vec: vector): vector`
+
+Applies `math.sign` component-wise for each vector. Equivalent of: `vector(math.sign(vec.x), ect)`.
+
+`vector.clamp(vec: vector, min: vector, max: vector): vector`
+
+Applies `math.clamp` component-wise for each vector. Equivalent of: `vector(math.clamp(vec.x, min.x, max.x), ect)`.
+
+`vector.max(...: vector): vector`
+
+Applies `math.max` component-wise for each vector. Equivalent of: `vector(math.max((...).x), ect)`.
+
+`vector.min(...: vector): vector`
+
+Applies `math.min` component-wise for each vector. Equivalent of: `vector(math.min((...).x), ect)`.
+
+---
+
+`vector.zero`
+
+Vector where `x=0, y=0, z=0, w?=0`.
+
+`vector.one`
+
+Vector where `x=1, y=1, z=1, w?=1`.
+
+---
+
+### Buffer Library
+
+`buffer.writevector(b: buffer, offset: number, vec: vector)`
+
+Writes the vector into the buffer at the offset. Each component will be written as a f32. In four component mode, it will write all four components and write 16 bytes. In three component mode, it will write 12 bytes.
+
+`buffer.readvector(b: buffer, offset: number): vector`
+
+Reads a vector from the buffer at the offset. In four component mode it will read 16 bytes and in three component mode it will read 12 bytes.
```

Comment:
This might not be the path for vector storage in buffers.
I would suggest taking a look at the discussion we had on this topic: https://github.com/luau-lang/luau/discussions/1296

I think a good path forward would be `readvector2`, `readvector3` and `readvector4` with similar methods to write.
I don't think f32xN is a good choice because it implies multiple value read/write unrelated to the vector type.

I do like the idea of having 2-component access as an option.

Reply:
functions removed and your idea has been placed into future work

Reply:
I can create a buffer update later when I go over pending buffer RFCs as well.